### PR TITLE
Allow to parse only the details needed in videodb

### PR DESCRIPTION
**Pull request by phate89**
With this changes we can ask only the needed details in GetFooInfo, GetFooNav and GetFoosByWhere. By default they behave like before.
I changed only where I was sure it was ok to have only partial details so it's used in json methods (GetFoos and GetFooDetails) and in a couple of places. I might have missed some place since GetFooInfo is used almost everywhere.

### CODE REVIEW

**Comment by Tolriq** [comment]
Maybe ping @Montellese as it's a follow up to discussion on previous patch for performances gains.

**Comment by Montellese** [comment]
Before looking through the whole PR: any specific reason you introduced a new enum `EDetailsFlag` instead of using the existing `Field` enum from `xbmc/utils/DatabaseUtils.h`?

File: xbmc/video/VideoDatabase.h [hunk 1]
```diff
@@ -67,6 +67,18 @@ namespace VIDEO
   struct SScanSettings;
 }
 
+enum EDetailsFlag
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.h]: We don't prefix enums with an `E`. And IMO `DetailsFlag` is a bit generic. Maybe something like
```
enum class VideoDbDetails
{
  None = 0x00,
  Rating = 0x01,
  ...
  All = 0xFF
};
```
That way you'll have to use `VideoDbDetails::None` etc which is IMO more obvious. And having `All` as `0xFF` is more future proof.

File: xbmc/video/VideoDatabase.cpp [hunk 2]
```diff
@@ -3843,14 +3840,15 @@ CVideoInfoTag CVideoDatabase::GetDetailsForMusicVideo(const dbiplus::sql_record*
 
   if (getDetails)
   {
-    GetTags(details.m_iDbId, MediaTypeMusicVideo, details.m_tags);
+    if (getDetails & CastDetail)
+      GetTags(details.m_iDbId, MediaTypeMusicVideo, details.m_tags);
 
     details.m_strPictureURL.Parse();
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp]: It's not obvious that this is done independent of the details being asked for.

File: xbmc/video/VideoDatabase.cpp [hunk 3]
```diff
@@ -3444,32 +3441,6 @@ void CVideoDatabase::GetDetailsFromDB(const dbiplus::sql_record* const record, i
 DWORD movieTime = 0;
 DWORD castTime = 0;
 
-CVideoInfoTag CVideoDatabase::GetDetailsByTypeAndId(VIDEODB_CONTENT_TYPE type, int id)
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp]: Please split this into a separate commit as it's not related to the optimization.

**Comment by Montellese** [comment]
Looks good apart from the comments. Not sure how best to handle the execution of `details.m_strPictureURL.Parse();` which is currently done whenever some detail is requested.

PS: Needs a rebase.

**Comment by phate89** [comment]
The parsing of pictureurl could be easily a new enum option.. that was my initial idea.. the problem is that I have no idea when it's needed. There are a lot of places where getfoonav or getfooinfo is called.. If someone knows better that part can easily add it

File: xbmc/video/VideoDatabase.h (hunk 1 above)
Comment (phate89) [inline xbmc/video/VideoDatabase.h]: To do it I need to also define bitmask operators for the class.. Is it worth it?

Comment (phate89) [inline xbmc/video/VideoDatabase.h]: @Montellese  With class you have to define bitmask operators to make it work.. Is it worth it or maybe just going with a better naming for it?

Comment (Montellese) [inline xbmc/video/VideoDatabase.h]: Then just change the naming to `VideoDbDetails` for the enum and `VideoDbDetailsNone` etc. for the enum values.

**Comment by phate89** [comment]
thanks @Montellese I addressed your comments. About parsing should I leave as is?

**Comment by Montellese** [comment]
What parsing?

**Comment by phate89** [comment]
The 'details.m_strPictureURL.Parse();'

**Comment by Montellese** [comment]
Ah right. I'd leave it as is for now. Otherwise it's getting messy.

**Comment by phate89** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
Builds are fine just the usual addon build crashes
